// thirdweb/dist/esm/react/web/ui/Bridge/payment-success/PaymentReceipt.js
"use client";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { CopyIcon } from "@radix-ui/react-icons";
import { useQuery } from "@tanstack/react-query";
import { useCallback } from "react";
import { defineChain, getCachedChain, getChainMetadata, } from "../../../../../chains/utils.js";
import { shortenHex } from "../../../../../utils/address.js";
import { formatExplorerTxUrl } from "../../../../../utils/url.js";
import { useCustomTheme } from "../../../../core/design-system/CustomThemeProvider.js";
import { iconSize, radius, spacing, } from "../../../../core/design-system/index.js";
import { formatTokenAmount } from "../../ConnectWallet/screens/formatTokenBalance.js";
import { shorterChainName } from "../../components/ChainName.js";
import { Skeleton } from "../../components/Skeleton.js";
import { Spacer } from "../../components/Spacer.js";
import { Container, ModalHeader } from "../../components/basic.js";
import { Text } from "../../components/text.js";
function getPaymentId(preparedQuote, status) {
    if (preparedQuote.type === "onramp") {
        return preparedQuote.id;
    }
    return status.transactions[status.transactions.length - 1]?.transactionHash;
}
/**
 * Hook to fetch transaction info for a completed status
 */
function useTransactionInfo(status, preparedQuote) {
    return useQuery({
        queryKey: [
            "transaction-info",
            status.type,
            getPaymentId(preparedQuote, status),
        ],
        queryFn: async () => {
            const isOnramp = status.type === "onramp";
            if (isOnramp && preparedQuote.type === "onramp") {
                // For onramp, create a display ID since OnrampStatus doesn't have paymentId
                return {
                    type: "paymentId",
                    id: preparedQuote.id,
                    label: "Onramp Payment",
                    destinationToken: preparedQuote.destinationToken,
                    chain: await getChainMetadata(defineChain(preparedQuote.destinationToken.chainId)),
                    amountPaid: `${preparedQuote.currencyAmount} ${preparedQuote.currency}`,
                    amountReceived: `${formatTokenAmount(preparedQuote.destinationAmount, preparedQuote.destinationToken.decimals)} ${preparedQuote.destinationToken.symbol}`,
                };
            }
            else if (status.type === "buy" ||
                status.type === "sell" ||
                status.type === "transfer") {
                if (status.transactions.length > 0) {
                    // get the last transaction hash
                    const tx = status.transactions[status.transactions.length - 1];
                    if (tx) {
                        const [destinationChain, originChain] = await Promise.all([
                            getChainMetadata(getCachedChain(status.destinationToken.chainId)),
                            getChainMetadata(getCachedChain(status.originToken.chainId)),
                        ]);
                        return {
                            type: "transactionHash",
                            id: tx.transactionHash,
                            label: "Transaction",
                            chain: destinationChain,
                            originToken: status.originToken,
                            originChain,
                            destinationToken: status.destinationToken,
                            amountReceived: `${formatTokenAmount(status.destinationAmount, status.destinationToken.decimals)} ${status.destinationToken.symbol}`,
                            amountPaid: `${formatTokenAmount(status.originAmount, status.originToken.decimals)} ${status.originToken.symbol}`,
                        };
                    }
                }
            }
            return null;
        },
        enabled: true,
        staleTime: 5 * 60 * 1000, // 5 minutes
    });
}
/**
 * Component to display details for a completed transaction step
 */
function CompletedStepDetailCard({ status, preparedQuote, windowAdapter, onCopyToClipboard, }) {
    const theme = useCustomTheme();
    const { data: txInfo, isLoading } = useTransactionInfo(status, preparedQuote);
    if (isLoading) {
        return (_jsxs(Container, { flex: "column", gap: "sm", style: {
                padding: spacing.md,
                borderRadius: radius.sm,
                backgroundColor: theme.colors.tertiaryBg,
                border: `1px solid ${theme.colors.borderColor}`,
            }, children: [_jsx(Skeleton, { height: "30px" }), _jsx(Skeleton, { height: "30px" }), _jsx(Skeleton, { height: "30px" })] }));
    }
    if (!txInfo) {
        return null;
    }
    return (_jsxs(Container, { flex: "column", gap: "sm", style: {
            padding: spacing.md,
            borderRadius: radius.sm,
            backgroundColor: theme.colors.tertiaryBg,
            border: `1px solid ${theme.colors.borderColor}`,
        }, children: [_jsxs(Container, { flex: "row", gap: "sm", style: {
                    alignItems: "center",
                    justifyContent: "space-between",
                }, children: [_jsx(Text, { size: "sm", color: "primaryText", children: txInfo.label }), _jsx(Container, { style: {
                            padding: `${spacing["3xs"]} ${spacing.xs}`,
                            borderRadius: radius.sm,
                            backgroundColor: theme.colors.success,
                        }, children: _jsx(Text, { size: "xs", style: { color: theme.colors.primaryButtonText }, children: "COMPLETED" }) })] }), txInfo.amountPaid && (_jsxs(Container, { flex: "row", center: "y", style: { justifyContent: "space-between" }, children: [_jsx(Text, { size: "sm", color: "secondaryText", children: "Amount Paid" }), _jsx(Text, { size: "sm", color: "primaryText", children: txInfo.amountPaid })] })), txInfo.originChain && (_jsxs(Container, { flex: "row", center: "y", style: { justifyContent: "space-between" }, children: [_jsx(Text, { size: "sm", color: "secondaryText", children: "Origin Chain" }), _jsx(Text, { size: "sm", color: "primaryText", children: shorterChainName(txInfo.chain.name) })] })), txInfo.amountReceived && (_jsxs(Container, { flex: "row", center: "y", style: { justifyContent: "space-between" }, children: [_jsx(Text, { size: "sm", color: "secondaryText", children: "Amount Received" }), _jsx(Text, { size: "sm", color: "primaryText", children: txInfo.amountReceived })] })), _jsxs(Container, { flex: "row", center: "y", style: { justifyContent: "space-between" }, children: [_jsx(Text, { size: "sm", color: "secondaryText", children: "Chain" }), _jsx(Text, { size: "sm", color: "primaryText", children: shorterChainName(txInfo.chain.name) })] }), _jsxs(Container, { flex: "row", center: "y", style: { justifyContent: "space-between" }, children: [_jsx(Text, { size: "sm", color: "secondaryText", children: txInfo.type === "paymentId" ? "Payment ID" : "Transaction Hash" }), _jsxs(Container, { flex: "row", gap: "sm", style: { alignItems: "center" }, children: [_jsx(Text, { size: "sm", color: "accentText", style: {
                                    fontFamily: "monospace",
                                    cursor: "pointer",
                                }, onClick: txInfo.type === "paymentId"
                                    ? () => onCopyToClipboard(txInfo.id)
                                    : () => {
                                        const explorer = txInfo.chain.explorers?.[0];
                                        if (explorer) {
                                            windowAdapter.open(formatExplorerTxUrl(explorer.url, txInfo.id));
                                        }
                                    }, children: shortenHex(txInfo.id) }), txInfo.type === "paymentId" ? (_jsx("button", { type: "button", style: {
                                    cursor: "pointer",
                                    background: "none",
                                    border: "none",
                                    padding: 0,
                                }, onClick: () => onCopyToClipboard(txInfo.id), children: _jsx(CopyIcon, { width: iconSize.sm, height: iconSize.sm, color: theme.colors.primaryText }) })) : null] })] })] }, txInfo.id));
}
export function PaymentReceipt({ preparedQuote, completedStatuses, onBack, windowAdapter, }) {
    // Copy to clipboard
    const copyToClipboard = useCallback(async (text) => {
        try {
            await navigator.clipboard.writeText(text);
            // Could add a toast notification here
        }
        catch (error) {
            console.warn("Failed to copy to clipboard:", error);
        }
    }, []);
    return (_jsxs(Container, { flex: "column", fullHeight: true, p: "lg", style: { maxHeight: "500px", minHeight: "250px", overflowY: "auto" }, children: [_jsx(ModalHeader, { title: "Payment Receipt", onBack: onBack }), _jsx(Spacer, { y: "lg" }), _jsx(Container, { flex: "column", gap: "lg", children: _jsxs(Container, { flex: "column", gap: "md", children: [_jsx(Text, { size: "md", color: "primaryText", children: "Transactions" }), completedStatuses.map((status, index) => (_jsx(CompletedStepDetailCard, { status: status, preparedQuote: preparedQuote, windowAdapter: windowAdapter, onCopyToClipboard: copyToClipboard }, `${status.type}-${index}`)))] }) })] }));
}
