// thirdweb/dist/esm/react/web/ui/Bridge/payment-success/SuccessScreen.js
"use client";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { CheckIcon } from "@radix-ui/react-icons";
import { useState } from "react";
import { useCustomTheme } from "../../../../core/design-system/CustomThemeProvider.js";
import { iconSize } from "../../../../core/design-system/index.js";
import { Spacer } from "../../components/Spacer.js";
import { Container, ModalHeader } from "../../components/basic.js";
import { Button } from "../../components/buttons.js";
import { Text } from "../../components/text.js";
import { PaymentReceipt } from "./PaymentReceipt.js";
export function SuccessScreen({ uiOptions, preparedQuote, completedStatuses, onDone, windowAdapter, }) {
    const theme = useCustomTheme();
    const [viewState, setViewState] = useState("success");
    if (viewState === "detail") {
        return (_jsx(PaymentReceipt, { preparedQuote: preparedQuote, completedStatuses: completedStatuses, windowAdapter: windowAdapter, onBack: () => setViewState("success") }));
    }
    return (_jsxs(Container, { flex: "column", fullHeight: true, p: "lg", children: [_jsx(ModalHeader, { title: "Payment Complete" }), _jsx(Spacer, { y: "xl" }), _jsxs(Container, { flex: "column", gap: "md", center: "x", children: [_jsx(Container, { center: "both", flex: "row", style: {
                            width: "64px",
                            height: "64px",
                            borderRadius: "50%",
                            backgroundColor: theme.colors.tertiaryBg,
                            marginBottom: "16px",
                            border: `2px solid ${theme.colors.success}`,
                            animation: "successBounce 0.6s ease-out",
                        }, children: _jsx(CheckIcon, { width: iconSize.xl, height: iconSize.xl, color: theme.colors.success, style: {
                                animation: "checkAppear 0.3s ease-out 0.3s both",
                            } }) }), _jsx(Text, { size: "xl", color: "primaryText", center: true, children: "Payment Successful!" }), _jsx(Text, { size: "sm", color: "secondaryText", center: true, children: "Your cross-chain payment has been completed successfully." })] }), _jsx(Spacer, { y: "lg" }), _jsxs(Container, { flex: "column", gap: "sm", style: { width: "100%" }, children: [_jsx(Button, { variant: "secondary", fullWidth: true, onClick: () => setViewState("detail"), children: "View Payment Receipt" }), _jsx(Button, { variant: "accent", fullWidth: true, onClick: onDone, children: uiOptions.mode === "transaction" ? "Continue" : "Done" })] }), _jsx("style", { children: `
          @keyframes successBounce {
            0% {
              transform: scale(0.3);
              opacity: 0;
            }
            50% {
              transform: scale(1.05);
            }
            70% {
              transform: scale(0.9);
            }
            100% {
              transform: scale(1);
              opacity: 1;
            }
          }

          @keyframes checkAppear {
            0% {
              transform: scale(0);
              opacity: 0;
            }
            100% {
              transform: scale(1);
              opacity: 1;
            }
          }
        ` })] }));
}
