// thirdweb/dist/cjs/react/web/ui/Bridge/BuyWidget.js
"use strict";
"use client";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuyWidget = BuyWidget;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_query_1 = require("@tanstack/react-query");
const addresses_js_1 = require("../../../../constants/addresses.js");
const get_token_js_1 = require("../../../../pay/convert/get-token.js");
const address_js_1 = require("../../../../utils/address.js");
const json_js_1 = require("../../../../utils/json.js");
const units_js_1 = require("../../../../utils/units.js");
const CustomThemeProvider_js_1 = require("../../../core/design-system/CustomThemeProvider.js");
const ConnectEmbed_js_1 = require("../ConnectWallet/Modal/ConnectEmbed.js");
const getConnectLocale_js_1 = require("../ConnectWallet/locale/getConnectLocale.js");
const DynamicHeight_js_1 = require("../components/DynamicHeight.js");
const Spinner_js_1 = require("../components/Spinner.js");
const BridgeOrchestrator_js_1 = require("./BridgeOrchestrator.js");
const UnsupportedTokenScreen_js_1 = require("./UnsupportedTokenScreen.js");
/**
 * Widget is a prebuilt UI for purchasing a specific token.
 *
 * @param props - Props of type [`BuyWidgetProps`](https://portal.thirdweb.com/references/typescript/v5/BuyWidgetProps) to configure the BuyWidget component.
 *
 * @example
 * ### Basic usage
 *
 * The `BuyWidget` component requires `client`, `chain`, and `amount` props to function.
 *
 * ```tsx
 * import { ethereum } from "thirdweb/chains";
 * import { toWei } from "thirdweb";
 *
 * <BuyWidget
 *   client={client}
 *   chain={ethereum}
 *   amount={toWei("0.1")}
 * />
 * ```
 *
 * ### Buy a specific token
 *
 * You can specify a token to purchase by passing the `tokenAddress` prop.
 *
 * ```tsx
 * <BuyWidget
 *   client={client}
 *   chain={ethereum}
 *   amount={toWei("100")}
 *   tokenAddress="0xA0b86a33E6417E4df2057B2d3C6d9F7cc11b0a70"
 * />
 * ```
 *
 * ### Customize the UI
 *
 * You can customize the UI of the `BuyWidget` component by passing a custom theme object to the `theme` prop.
 *
 * ```tsx
 * <BuyWidget
 *   client={client}
 *   chain={ethereum}
 *   amount={toWei("0.1")}
 *   theme={darkTheme({
 *     colors: {
 *       modalBg: "red",
 *     },
 *   })}
 * />
 * ```
 *
 * Refer to the [`Theme`](https://portal.thirdweb.com/references/typescript/v5/Theme) type for more details.
 *
 * ### Update the Title
 *
 * You can update the title of the widget by passing a `title` prop to the `BuyWidget` component.
 *
 * ```tsx
 * <BuyWidget
 *   client={client}
 *   chain={ethereum}
 *   amount={toWei("0.1")}
 *   title="Buy ETH"
 * />
 * ```
 *
 * ### Configure the wallet connection
 *
 * You can customize the wallet connection flow by passing a `connectOptions` object to the `BuyWidget` component.
 *
 * ```tsx
 * <BuyWidget
 *   client={client}
 *   chain={ethereum}
 *   amount={toWei("0.1")}
 *   connectOptions={{
 *     connectModal: {
 *       size: 'compact',
 *       title: "Sign in",
 *     }
 *   }}
 * />
 * ```
 *
 * Refer to the [`BuyWidgetConnectOptions`](https://portal.thirdweb.com/references/typescript/v5/BuyWidgetConnectOptions) type for more details.
 *
 * @bridge
 * @beta
 * @react
 */
function BuyWidget(props) {
    const localeQuery = (0, getConnectLocale_js_1.useConnectLocale)(props.locale || "en_US");
    const theme = props.theme || "dark";
    const bridgeDataQuery = (0, react_query_1.useQuery)({
        queryKey: ["bridgeData", (0, json_js_1.stringify)(props)],
        queryFn: async () => {
            if (!props.tokenAddress ||
                ((0, address_js_1.isAddress)(props.tokenAddress) &&
                    (0, address_js_1.checksumAddress)(props.tokenAddress) ===
                        (0, address_js_1.checksumAddress)(addresses_js_1.NATIVE_TOKEN_ADDRESS))) {
                const ETH = await (0, get_token_js_1.getToken)(props.client, addresses_js_1.NATIVE_TOKEN_ADDRESS, props.chain.id);
                return {
                    type: "success",
                    data: {
                        mode: "fund_wallet",
                        destinationToken: ETH,
                        initialAmount: (0, units_js_1.toTokens)(props.amount, ETH.decimals),
                    },
                };
            }
            const token = await (0, get_token_js_1.getToken)(props.client, props.tokenAddress, props.chain.id).catch((err) => {
                err.message.includes("not supported") ? undefined : Promise.reject(err);
            });
            if (!token) {
                return {
                    type: "unsupported_token",
                    tokenAddress: props.tokenAddress,
                    chain: props.chain,
                };
            }
            return {
                type: "success",
                data: {
                    mode: "fund_wallet",
                    destinationToken: token,
                    initialAmount: (0, units_js_1.toTokens)(props.amount, token.decimals),
                    metadata: {
                        title: props.title,
                    },
                },
            };
        },
    });
    let content = null;
    if (!localeQuery.data || bridgeDataQuery.isLoading) {
        content = ((0, jsx_runtime_1.jsx)("div", { style: {
                minHeight: "350px",
                display: "flex",
                justifyContent: "center",
                alignItems: "center",
            }, children: (0, jsx_runtime_1.jsx)(Spinner_js_1.Spinner, { size: "xl", color: "secondaryText" }) }));
    }
    else if (bridgeDataQuery.data?.type === "unsupported_token") {
        // Show unsupported token screen
        content = (0, jsx_runtime_1.jsx)(UnsupportedTokenScreen_js_1.UnsupportedTokenScreen, { chain: bridgeDataQuery.data.chain });
    }
    else if (bridgeDataQuery.data?.type === "success") {
        // Show normal bridge orchestrator
        content = ((0, jsx_runtime_1.jsx)(BridgeOrchestrator_js_1.BridgeOrchestrator, { client: props.client, uiOptions: bridgeDataQuery.data.data, connectOptions: props.connectOptions, connectLocale: localeQuery.data, purchaseData: props.purchaseData, paymentLinkId: props.paymentLinkId, onComplete: () => {
                props.onSuccess?.();
            }, onError: (err) => {
                props.onError?.(err);
            }, onCancel: () => {
                props.onCancel?.();
            }, presetOptions: props.presetOptions, receiverAddress: undefined }));
    }
    return ((0, jsx_runtime_1.jsx)(CustomThemeProvider_js_1.CustomThemeProvider, { theme: theme, children: (0, jsx_runtime_1.jsx)(ConnectEmbed_js_1.EmbedContainer, { modalSize: "compact", style: props.style, className: props.className, children: (0, jsx_runtime_1.jsx)(DynamicHeight_js_1.DynamicHeight, { children: content }) }) }));
}
