// thirdweb/dist/esm/react/web/ui/Bridge/payment-selection/WalletFiatSelection.js
"use client";
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { ChevronRightIcon, PlusIcon } from "@radix-ui/react-icons";
import { useCustomTheme } from "../../../../core/design-system/CustomThemeProvider.js";
import { iconSize, radius, spacing, } from "../../../../core/design-system/index.js";
import { CreditCardIcon } from "../../ConnectWallet/icons/CreditCardIcon.js";
import { WalletRow } from "../../ConnectWallet/screens/Buy/swap/WalletRow.js";
import { Spacer } from "../../components/Spacer.js";
import { Container } from "../../components/basic.js";
import { Button } from "../../components/buttons.js";
import { Text } from "../../components/text.js";
export function WalletFiatSelection({ connectedWallets, client, onWalletSelected, onFiatSelected, onConnectWallet, }) {
    const theme = useCustomTheme();
    return (_jsxs(_Fragment, { children: [_jsx(Text, { size: "md", color: "primaryText", children: "Pay with Crypto" }), _jsx(Spacer, { y: "md" }), connectedWallets.length > 0 && (_jsxs(_Fragment, { children: [_jsx(Container, { flex: "column", gap: "sm", children: connectedWallets.map((wallet) => {
                            const account = wallet.getAccount();
                            if (!account?.address) {
                                return null;
                            }
                            return (_jsxs(Button, { variant: "secondary", fullWidth: true, onClick: () => onWalletSelected(wallet), style: {
                                    border: `1px solid ${theme.colors.borderColor}`,
                                    borderRadius: radius.md,
                                    padding: `${spacing.sm} ${spacing.md}`,
                                    backgroundColor: theme.colors.tertiaryBg,
                                    justifyContent: "space-between",
                                }, children: [_jsx(WalletRow, { client: client, address: account?.address, iconSize: "lg", textSize: "sm" }), _jsx(ChevronRightIcon, { style: { width: iconSize.md, height: iconSize.md } })] }, wallet.id));
                        }) }), _jsx(Spacer, { y: "sm" })] })), _jsx(Button, { variant: "secondary", fullWidth: true, onClick: onConnectWallet, style: {
                    border: `1px solid ${theme.colors.borderColor}`,
                    borderRadius: radius.md,
                    padding: `${spacing.sm} ${spacing.md}`,
                    backgroundColor: theme.colors.tertiaryBg,
                    textAlign: "left",
                    height: "auto",
                }, children: _jsxs(Container, { flex: "row", gap: "md", style: { width: "100%", alignItems: "center" }, children: [_jsx(Container, { style: {
                                borderRadius: radius.sm,
                                border: `1px dashed ${theme.colors.secondaryIconColor}`,
                                display: "flex",
                                alignItems: "center",
                                justifyContent: "center",
                                padding: spacing["4xs"],
                                width: iconSize.lg,
                                height: iconSize.lg,
                            }, children: _jsx(PlusIcon, { width: iconSize.md, height: iconSize.md, color: theme.colors.secondaryText }) }), _jsxs(Container, { flex: "column", gap: "3xs", style: { flex: 1 }, children: [_jsx(Text, { size: "sm", color: "primaryText", style: { fontWeight: 600 }, children: "Connect Another Wallet" }), _jsx(Text, { size: "xs", color: "secondaryText", children: "Use a different wallet to pay" })] })] }) }), _jsx(Spacer, { y: "md" }), _jsx(Text, { size: "md", color: "primaryText", children: "Pay with Fiat" }), _jsx(Spacer, { y: "md" }), _jsx(Button, { variant: "secondary", fullWidth: true, onClick: onFiatSelected, style: {
                    border: `1px solid ${theme.colors.borderColor}`,
                    borderRadius: radius.md,
                    padding: `${spacing.sm} ${spacing.md}`,
                    backgroundColor: theme.colors.tertiaryBg,
                    textAlign: "left",
                    height: "auto",
                }, children: _jsxs(Container, { flex: "row", gap: "md", style: { width: "100%", alignItems: "center" }, children: [_jsx(CreditCardIcon, { size: iconSize.lg, color: theme.colors.secondaryIconColor }), _jsxs(Container, { flex: "column", gap: "3xs", style: { flex: 1 }, children: [_jsx(Text, { size: "sm", color: "primaryText", style: { fontWeight: 600 }, children: "Pay with Card" }), _jsx(Text, { size: "xs", color: "secondaryText", children: "Buy crypto and bridge in one step" })] })] }) })] }));
}
