// thirdweb/dist/cjs/react/web/ui/Bridge/payment-success/SuccessScreen.js
"use strict";
"use client";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuccessScreen = SuccessScreen;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_icons_1 = require("@radix-ui/react-icons");
const react_1 = require("react");
const CustomThemeProvider_js_1 = require("../../../../core/design-system/CustomThemeProvider.js");
const index_js_1 = require("../../../../core/design-system/index.js");
const Spacer_js_1 = require("../../components/Spacer.js");
const basic_js_1 = require("../../components/basic.js");
const buttons_js_1 = require("../../components/buttons.js");
const text_js_1 = require("../../components/text.js");
const PaymentReceipt_js_1 = require("./PaymentReceipt.js");
function SuccessScreen({ uiOptions, preparedQuote, completedStatuses, onDone, windowAdapter, }) {
    const theme = (0, CustomThemeProvider_js_1.useCustomTheme)();
    const [viewState, setViewState] = (0, react_1.useState)("success");
    if (viewState === "detail") {
        return ((0, jsx_runtime_1.jsx)(PaymentReceipt_js_1.PaymentReceipt, { preparedQuote: preparedQuote, completedStatuses: completedStatuses, windowAdapter: windowAdapter, onBack: () => setViewState("success") }));
    }
    return ((0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", fullHeight: true, p: "lg", children: [(0, jsx_runtime_1.jsx)(basic_js_1.ModalHeader, { title: "Payment Complete" }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "xl" }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", gap: "md", center: "x", children: [(0, jsx_runtime_1.jsx)(basic_js_1.Container, { center: "both", flex: "row", style: {
                            width: "64px",
                            height: "64px",
                            borderRadius: "50%",
                            backgroundColor: theme.colors.tertiaryBg,
                            marginBottom: "16px",
                            border: `2px solid ${theme.colors.success}`,
                            animation: "successBounce 0.6s ease-out",
                        }, children: (0, jsx_runtime_1.jsx)(react_icons_1.CheckIcon, { width: index_js_1.iconSize.xl, height: index_js_1.iconSize.xl, color: theme.colors.success, style: {
                                animation: "checkAppear 0.3s ease-out 0.3s both",
                            } }) }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "xl", color: "primaryText", center: true, children: "Payment Successful!" }), (0, jsx_runtime_1.jsx)(text_js_1.Text, { size: "sm", color: "secondaryText", center: true, children: "Your cross-chain payment has been completed successfully." })] }), (0, jsx_runtime_1.jsx)(Spacer_js_1.Spacer, { y: "lg" }), (0, jsx_runtime_1.jsxs)(basic_js_1.Container, { flex: "column", gap: "sm", style: { width: "100%" }, children: [(0, jsx_runtime_1.jsx)(buttons_js_1.Button, { variant: "secondary", fullWidth: true, onClick: () => setViewState("detail"), children: "View Payment Receipt" }), (0, jsx_runtime_1.jsx)(buttons_js_1.Button, { variant: "accent", fullWidth: true, onClick: onDone, children: uiOptions.mode === "transaction" ? "Continue" : "Done" })] }), (0, jsx_runtime_1.jsx)("style", { children: `
          @keyframes successBounce {
            0% {
              transform: scale(0.3);
              opacity: 0;
            }
            50% {
              transform: scale(1.05);
            }
            70% {
              transform: scale(0.9);
            }
            100% {
              transform: scale(1);
              opacity: 1;
            }
          }

          @keyframes checkAppear {
            0% {
              transform: scale(0);
              opacity: 0;
            }
            100% {
              transform: scale(1);
              opacity: 1;
            }
          }
        ` })] }));
}
