// thirdweb/dist/esm/react/web/ui/Bridge/ErrorBanner.js
"use client";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { CrossCircledIcon } from "@radix-ui/react-icons";
import { useCustomTheme } from "../../../core/design-system/CustomThemeProvider.js";
import { iconSize } from "../../../core/design-system/index.js";
import { useBridgeError } from "../../../core/hooks/useBridgeError.js";
import { Container } from "../components/basic.js";
import { Button } from "../components/buttons.js";
import { Text } from "../components/text.js";
export function ErrorBanner({ error, onRetry, onCancel }) {
    const theme = useCustomTheme();
    const { userMessage } = useBridgeError({ error });
    return (_jsx(Container, { flex: "column", gap: "md", p: "lg", fullHeight: true, children: _jsxs(Container, { flex: "row", gap: "md", style: { alignItems: "flex-start" }, children: [_jsx(Container, { center: "both", style: {
                        width: "24px",
                        height: "24px",
                        borderRadius: "50%",
                        backgroundColor: theme.colors.tertiaryBg,
                        flexShrink: 0,
                    }, children: _jsx(CrossCircledIcon, { width: iconSize.md, height: iconSize.md, color: theme.colors.danger }) }), _jsxs(Container, { flex: "column", gap: "sm", style: { flex: 1 }, fullHeight: true, children: [_jsx(Text, { size: "lg", color: "primaryText", children: "Error" }), _jsx(Container, { flex: "column", gap: "sm", style: {
                                minHeight: "100px",
                            }, children: _jsx(Container, { flex: "column", gap: "sm", style: { flex: 1 }, children: _jsx(Text, { size: "sm", color: "secondaryText", children: userMessage }) }) }), _jsxs(Container, { flex: "row", gap: "sm", style: { justifyContent: "flex-end" }, children: [_jsx(Button, { variant: "primary", onClick: onRetry, children: "Try Again" }), onCancel && (_jsx(Button, { variant: "secondary", onClick: onCancel, children: "Cancel" }))] })] })] }) }));
}
