// thirdweb/dist/esm/react/web/ui/Bridge/StepRunner.js
"use client";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { CheckIcon, ClockIcon, Cross1Icon } from "@radix-ui/react-icons";
import { defineChain } from "../../../../chains/utils.js";
import { useCustomTheme } from "../../../core/design-system/CustomThemeProvider.js";
import { iconSize, radius, spacing, } from "../../../core/design-system/index.js";
import { useStepExecutor, } from "../../../core/hooks/useStepExecutor.js";
import { ChainName } from "../components/ChainName.js";
import { Spacer } from "../components/Spacer.js";
import { Spinner } from "../components/Spinner.js";
import { Container, ModalHeader } from "../components/basic.js";
import { Button } from "../components/buttons.js";
import { Text } from "../components/text.js";
export function StepRunner({ request, wallet, client, windowAdapter, onComplete, onCancel, onBack, autoStart, }) {
    const theme = useCustomTheme();
    // Use the real step executor hook
    const { currentStep, progress, executionState, onrampStatus, steps, error, start, cancel, retry, } = useStepExecutor({
        request,
        wallet,
        client,
        windowAdapter,
        autoStart,
        onComplete: (completedStatuses) => {
            onComplete(completedStatuses);
        },
    });
    const handleCancel = () => {
        cancel();
        if (onCancel) {
            onCancel();
        }
    };
    const handleRetry = () => {
        retry();
    };
    const getStepStatus = (stepIndex) => {
        if (!currentStep || !steps) {
            // Not started yet
            return stepIndex === 0 ? (error ? "failed" : "pending") : "pending";
        }
        const currentStepIndex = steps.findIndex((step) => step === currentStep);
        if (stepIndex < currentStepIndex)
            return "completed";
        if (stepIndex === currentStepIndex && executionState === "executing")
            return "executing";
        if (stepIndex === currentStepIndex && error)
            return "failed";
        if (stepIndex === currentStepIndex &&
            executionState === "idle" &&
            progress === 100)
            return "completed";
        return "pending";
    };
    const getStatusIcon = (status) => {
        switch (status) {
            case "completed":
                return (_jsx(CheckIcon, { width: iconSize.sm, height: iconSize.sm, color: theme.colors.accentButtonText }));
            case "executing":
                return _jsx(Spinner, { size: "sm", color: "accentButtonText" });
            case "failed":
                return (_jsx(Cross1Icon, { width: iconSize.sm, height: iconSize.sm, color: "white" }));
            default:
                return (_jsx(ClockIcon, { width: iconSize.sm, height: iconSize.sm, color: theme.colors.primaryText }));
        }
    };
    const getStepBackgroundColor = (status) => {
        switch (status) {
            case "completed":
                return theme.colors.tertiaryBg;
            case "executing":
                return theme.colors.tertiaryBg;
            case "failed":
                return theme.colors.tertiaryBg;
            default:
                return theme.colors.tertiaryBg;
        }
    };
    const getIconBackgroundColor = (status) => {
        switch (status) {
            case "completed":
                return theme.colors.success;
            case "executing":
                return theme.colors.accentButtonBg;
            case "failed":
                return theme.colors.danger;
            default:
                return theme.colors.borderColor;
        }
    };
    const getStepDescription = (step) => {
        const { originToken, destinationToken } = step;
        // If tokens are the same, it's likely a bridge operation
        if (originToken.chainId !== destinationToken.chainId) {
            return (_jsxs(Container, { flex: "row", gap: "3xs", children: [_jsxs(Text, { size: "sm", color: "primaryText", children: ["Bridge ", originToken.symbol, " to", " "] }), _jsx(ChainName, { chain: getDestinationChain(request), size: "sm", client: client, short: true, color: "primaryText" })] }));
        }
        // If different tokens on same chain, it's a swap
        if (originToken.symbol !== destinationToken.symbol) {
            return (_jsxs(Text, { size: "sm", color: "primaryText", children: ["Swap ", originToken.symbol, " to ", destinationToken.symbol] }));
        }
        // Fallback to step number
        return (_jsx(Text, { size: "sm", color: "primaryText", children: "Process transaction" }));
    };
    const getStepStatusText = (status) => {
        switch (status) {
            case "executing":
                return "Processing...";
            case "completed":
                return "Completed";
            case "pending":
                return "Waiting...";
            case "failed":
                return "Failed";
            default:
                return "Unknown";
        }
    };
    return (_jsxs(Container, { flex: "column", fullHeight: true, p: "lg", children: [_jsx(ModalHeader, { title: "Processing Payment", onBack: onBack }), _jsx(Spacer, { y: "xl" }), _jsxs(Container, { flex: "column", children: [_jsxs(Container, { flex: "column", gap: "sm", children: [_jsxs(Container, { flex: "row", center: "y", children: [_jsx(Text, { size: "sm", color: "secondaryText", style: { flex: 1 }, children: "Progress" }), _jsxs(Text, { size: "sm", color: "primaryText", children: [progress, "%"] })] }), _jsx(Container, { style: {
                                    width: "100%",
                                    height: "8px",
                                    backgroundColor: theme.colors.tertiaryBg,
                                    border: `1px solid ${theme.colors.borderColor}`,
                                    borderRadius: "4px",
                                    overflow: "hidden",
                                }, children: _jsx(Container, { style: {
                                        width: `${progress}%`,
                                        height: "100%",
                                        backgroundColor: error
                                            ? theme.colors.danger
                                            : theme.colors.accentButtonBg,
                                        transition: "width 0.3s ease",
                                    }, children: _jsx(Text, {}) }) })] }), _jsx(Spacer, { y: "lg" }), _jsxs(Container, { flex: "column", gap: "sm", children: [request.type === "onramp" && onrampStatus ? (_jsxs(Container, { flex: "row", gap: "md", style: {
                                    alignItems: "center",
                                    padding: `${spacing.sm} ${spacing.md}`,
                                    borderRadius: radius.md,
                                    backgroundColor: getStepBackgroundColor(onrampStatus),
                                    border: `1px solid ${theme.colors.borderColor}`,
                                }, children: [_jsx(Container, { center: "both", flex: "row", style: {
                                            width: `${iconSize.lg}px`,
                                            height: `${iconSize.lg}px`,
                                            borderRadius: "50%",
                                            backgroundColor: getIconBackgroundColor(onrampStatus),
                                            color: theme.colors.primaryButtonText,
                                            flexShrink: 0,
                                        }, children: getStatusIcon(onrampStatus) }), _jsxs(Container, { flex: "column", gap: "3xs", style: { flex: 1 }, children: [_jsx(Text, { size: "sm", color: "primaryText", children: "TEST" }), _jsx(Text, { size: "xs", color: "secondaryText", children: getStepStatusText(onrampStatus) })] })] })) : null, steps?.map((step, index) => {
                                const status = getStepStatus(index);
                                return (_jsxs(Container, { flex: "row", gap: "md", style: {
                                        alignItems: "center",
                                        padding: `${spacing.sm} ${spacing.md}`,
                                        borderRadius: radius.md,
                                        backgroundColor: getStepBackgroundColor(status),
                                        border: `1px solid ${theme.colors.borderColor}`,
                                    }, children: [_jsx(Container, { center: "both", flex: "row", style: {
                                                width: `${iconSize.lg}px`,
                                                height: `${iconSize.lg}px`,
                                                borderRadius: "50%",
                                                backgroundColor: getIconBackgroundColor(status),
                                                color: theme.colors.primaryButtonText,
                                                flexShrink: 0,
                                            }, children: getStatusIcon(status) }), _jsxs(Container, { flex: "column", gap: "3xs", style: { flex: 1 }, children: [getStepDescription(step), _jsx(Text, { size: "xs", color: "secondaryText", children: getStepStatusText(status) })] })] }, `${step.originToken.chainId}-${step.destinationToken.chainId}-${index}`));
                            })] }), _jsx(Spacer, { y: "md" }), _jsxs(Text, { size: "xs", color: "secondaryText", center: true, children: ["Keep this window open until all", _jsx("br", {}), " transactions are complete."] }), _jsx(Spacer, { y: "lg" }), error ? (_jsx(Container, { flex: "row", gap: "md", children: _jsx(Button, { variant: "primary", fullWidth: true, onClick: handleRetry, children: "Retry" }) })) : executionState === "idle" && progress === 0 ? (_jsx(Button, { variant: "accent", fullWidth: true, onClick: start, children: "Start Transaction" })) : executionState === "executing" ||
                        executionState === "auto-starting" ? (_jsx(Button, { variant: "secondary", fullWidth: true, onClick: handleCancel, children: "Cancel Transaction" })) : null] })] }));
}
function getDestinationChain(request) {
    switch (request.type) {
        case "onramp":
            return defineChain(request.chainId);
        case "buy":
        case "sell":
            return defineChain(request.destinationChainId);
        case "transfer":
            return defineChain(request.chainId);
        default:
            throw new Error("Invalid quote type");
    }
}
