// thirdweb/dist/cjs/stories/Bridge/ErrorBanner.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmptyMessageLight = exports.EmptyMessage = exports.WithoutCancelButtonLight = exports.WithoutCancelButton = exports.InsufficientFundsErrorLight = exports.InsufficientFundsError = exports.UserRejectedErrorLight = exports.UserRejectedError = exports.NetworkErrorLight = exports.NetworkError = exports.Dark = exports.Light = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const ErrorBanner_js_1 = require("../../react/web/ui/Bridge/ErrorBanner.js");
const utils_js_1 = require("../utils.js");
const mockNetworkError = new Error("Network connection failed. Please check your internet connection and try again.");
const mockUserRejectedError = new Error("Transaction was rejected by user.");
const mockInsufficientFundsError = new Error("Insufficient funds to complete this transaction.");
const mockGenericError = new Error("An unexpected error occurred.");
// Wrapper component to provide theme context
const ErrorBannerWithTheme = (props) => {
    const { theme, ...componentProps } = props;
    return ((0, jsx_runtime_1.jsx)(utils_js_1.ModalThemeWrapper, { theme: theme, children: (0, jsx_runtime_1.jsx)(ErrorBanner_js_1.ErrorBanner, { ...componentProps }) }));
};
const meta = {
    title: "Bridge/ErrorBanner",
    component: ErrorBannerWithTheme,
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "Error banner component that displays user-friendly error messages with retry functionality and optional cancel action.",
            },
        },
    },
    tags: ["autodocs"],
    args: {
        error: mockNetworkError,
        onRetry: () => { },
        onCancel: () => { },
        theme: "dark",
    },
    argTypes: {
        theme: {
            control: "select",
            options: ["light", "dark"],
            description: "Theme for the component",
        },
        onRetry: { action: "retry clicked" },
        onCancel: { action: "cancel clicked" },
    },
};
exports.default = meta;
exports.Light = {
    args: {
        theme: "light",
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.Dark = {
    args: {
        theme: "dark",
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.NetworkError = {
    args: {
        theme: "dark",
        error: mockNetworkError,
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.NetworkErrorLight = {
    args: {
        theme: "light",
        error: mockNetworkError,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.UserRejectedError = {
    args: {
        theme: "dark",
        error: mockUserRejectedError,
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.UserRejectedErrorLight = {
    args: {
        theme: "light",
        error: mockUserRejectedError,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.InsufficientFundsError = {
    args: {
        theme: "dark",
        error: mockInsufficientFundsError,
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.InsufficientFundsErrorLight = {
    args: {
        theme: "light",
        error: mockInsufficientFundsError,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.WithoutCancelButton = {
    args: {
        theme: "dark",
        error: mockGenericError,
        onCancel: undefined,
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.WithoutCancelButtonLight = {
    args: {
        theme: "light",
        error: mockGenericError,
        onCancel: undefined,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.EmptyMessage = {
    args: {
        theme: "dark",
        error: new Error(""),
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.EmptyMessageLight = {
    args: {
        theme: "light",
        error: new Error(""),
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
