// thirdweb/dist/cjs/stories/Bridge/PaymentSelection.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LargeAmount = exports.DifferentDestinationToken = exports.WithBackButtonLight = exports.WithBackButton = exports.Dark = exports.Light = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const PaymentSelection_js_1 = require("../../react/web/ui/Bridge/payment-selection/PaymentSelection.js");
const en_js_1 = require("../../react/web/ui/ConnectWallet/locale/en.js");
const utils_js_1 = require("../utils.js");
const fixtures_js_1 = require("./fixtures.js");
// Wrapper component to provide theme context
const PaymentSelectionWithTheme = (props) => {
    const { theme, ...componentProps } = props;
    return ((0, jsx_runtime_1.jsx)(utils_js_1.ModalThemeWrapper, { theme: theme, children: (0, jsx_runtime_1.jsx)(PaymentSelection_js_1.PaymentSelection, { ...componentProps }) }));
};
const meta = {
    title: "Bridge/PaymentSelection",
    component: PaymentSelectionWithTheme,
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "Payment method selection screen with a 2-step flow:\n\n" +
                    "**Step 1:** Choose payment method - shows connected wallets, connect wallet option, and pay with fiat option\n\n" +
                    "**Step 2a:** If wallet selected - shows available origin tokens for bridging to the destination token (fetches real routes data from the Bridge API)\n\n" +
                    "**Step 2b:** If fiat selected - shows onramp provider options (Coinbase, Stripe, Transak)\n\n" +
                    "The component intelligently manages wallet context and provides proper error handling for each step.",
            },
        },
    },
    tags: ["autodocs"],
    args: {
        destinationToken: fixtures_js_1.USDC,
        client: utils_js_1.storyClient,
        onPaymentMethodSelected: (_paymentMethod) => { },
        onError: (error) => console.error("Error:", error),
        theme: "dark",
        destinationAmount: "1",
        connectLocale: en_js_1.default,
    },
    argTypes: {
        theme: {
            control: "select",
            options: ["light", "dark"],
            description: "Theme for the component",
        },
        destinationToken: {
            description: "The target token to bridge to",
        },
        destinationAmount: {
            description: "Amount of destination token to bridge",
        },
        onPaymentMethodSelected: {
            action: "payment method selected",
            description: "Called when user selects a wallet token or fiat provider",
        },
        onError: {
            action: "error occurred",
            description: "Called when an error occurs during the flow",
        },
        onBack: {
            action: "back clicked",
            description: "Called when user wants to go back (only shown in Step 1)",
        },
        connectLocale: {
            description: "Locale for connecting wallets",
        },
    },
};
exports.default = meta;
exports.Light = {
    args: {
        theme: "light",
    },
    parameters: {
        backgrounds: { default: "light" },
        docs: {
            description: {
                story: "Light theme version showing the initial wallet selection step. Click on a connected wallet to see token selection, or click 'Pay with Fiat' to see provider selection.",
            },
        },
    },
};
exports.Dark = {
    args: {
        theme: "dark",
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Dark theme version of the payment selection flow. The component starts with wallet selection and provides navigation through the 2-step process.",
            },
        },
    },
};
exports.WithBackButton = {
    args: {
        theme: "dark",
        onBack: () => { },
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Version with a back button in the header. The back behavior changes based on the current step - Step 1 calls onBack, Steps 2a/2b return to Step 1.",
            },
        },
    },
};
exports.WithBackButtonLight = {
    args: {
        theme: "light",
        onBack: () => { },
    },
    parameters: {
        backgrounds: { default: "light" },
        docs: {
            description: {
                story: "Light theme version with back button functionality. Demonstrates the navigation flow between steps.",
            },
        },
    },
};
exports.DifferentDestinationToken = {
    args: {
        theme: "dark",
        destinationToken: fixtures_js_1.UNI,
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Example with a different destination token (UNI). This will show different available origin tokens in Step 2a when a wallet is selected.",
            },
        },
    },
};
exports.LargeAmount = {
    args: {
        theme: "dark",
        destinationAmount: "1000",
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Example with a larger destination amount (1000 USDC). This may affect which origin tokens are available based on user balances.",
            },
        },
    },
};
