// thirdweb/dist/cjs/stories/TokenBalanceRow.stories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DarkTokenList = exports.TokenList = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const ethereum_js_1 = require("../chains/chain-definitions/ethereum.js");
const CustomThemeProvider_js_1 = require("../react/core/design-system/CustomThemeProvider.js");
const TokenBalanceRow_js_1 = require("../react/web/ui/Bridge/common/TokenBalanceRow.js");
const fixtures_js_1 = require("./Bridge/fixtures.js");
const utils_js_1 = require("./utils.js");
const dummyBalanceETH = "1.2345";
const dummyBalanceUSDC = "1234.56";
const dummyBalanceLowUNI = "0.0012";
// Wrapper component to provide theme context
const TokenBalanceRowWithTheme = (props) => {
    const { theme, ...tokenBalanceRowProps } = props;
    return ((0, jsx_runtime_1.jsx)(CustomThemeProvider_js_1.CustomThemeProvider, { theme: theme, children: (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { ...tokenBalanceRowProps }) }));
};
const meta = {
    title: "Bridge/TokenBalanceRow",
    component: TokenBalanceRowWithTheme,
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "A row component that displays token balance information including token icon, symbol, chain, balance amount and fiat value. Used in bridge interfaces for token selection.",
            },
        },
    },
    tags: ["autodocs"],
    args: {
        client: utils_js_1.storyClient,
        token: fixtures_js_1.ETH,
        chain: ethereum_js_1.ethereum,
        amount: dummyBalanceETH,
        onClick: (_token) => { },
        theme: "dark",
    },
    argTypes: {
        theme: {
            control: "select",
            options: ["light", "dark"],
            description: "Theme for the component",
        },
        onClick: {
            action: "clicked",
            description: "Callback function when token row is clicked",
        },
    },
};
exports.TokenList = {
    render: (args) => ((0, jsx_runtime_1.jsx)(CustomThemeProvider_js_1.CustomThemeProvider, { theme: args.theme, children: (0, jsx_runtime_1.jsxs)("div", { style: {
                display: "flex",
                flexDirection: "column",
                gap: "8px",
                minWidth: "400px",
                maxWidth: "500px",
            }, children: [(0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.ETH, amount: dummyBalanceETH, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.ETH, amount: dummyBalanceETH, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.USDC, amount: dummyBalanceUSDC, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.UNI, amount: dummyBalanceLowUNI, onClick: args.onClick })] }) })),
    args: {
        theme: "light",
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
exports.DarkTokenList = {
    render: (args) => ((0, jsx_runtime_1.jsx)(CustomThemeProvider_js_1.CustomThemeProvider, { theme: args.theme, children: (0, jsx_runtime_1.jsxs)("div", { style: {
                display: "flex",
                flexDirection: "column",
                gap: "8px",
                minWidth: "400px",
                maxWidth: "500px",
            }, children: [(0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.ETH, amount: dummyBalanceETH, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.ETH, amount: dummyBalanceETH, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.USDC, amount: dummyBalanceUSDC, onClick: args.onClick }), (0, jsx_runtime_1.jsx)(TokenBalanceRow_js_1.TokenBalanceRow, { client: args.client, token: fixtures_js_1.UNI, amount: dummyBalanceLowUNI, onClick: args.onClick })] }) })),
    args: {
        theme: "dark",
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
exports.default = meta;
