// thirdweb/dist/esm/stories/Bridge/SuccessScreen.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { stringify } from "viem";
import { webWindowAdapter } from "../../react/web/adapters/WindowAdapter.js";
import { SuccessScreen, } from "../../react/web/ui/Bridge/payment-success/SuccessScreen.js";
import { ModalThemeWrapper } from "../utils.js";
import { FUND_WALLET_UI_OPTIONS, TRANSACTION_UI_OPTIONS, simpleBuyQuote, simpleOnrampQuote, } from "./fixtures.js";
const mockBuyCompletedStatuses = JSON.parse(stringify([
    {
        type: "buy",
        status: "COMPLETED",
        paymentId: "payment-12345",
        originAmount: 1000000000000000000n,
        destinationAmount: 100000000n,
        originChainId: 1,
        destinationChainId: 1,
        originTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        destinationTokenAddress: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
        originToken: {
            chainId: 1,
            address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
            symbol: "ETH",
            name: "Ethereum",
            decimals: 18,
            priceUsd: 2500,
        },
        destinationToken: {
            chainId: 1,
            address: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
            symbol: "USDC",
            name: "USD Coin",
            decimals: 6,
            priceUsd: 1,
        },
        sender: "0xa3841994009B4fEabb01ebcC62062F9E56F701CD",
        receiver: "0xa3841994009B4fEabb01ebcC62062F9E56F701CD",
        transactions: [
            {
                chainId: 1,
                transactionHash: "0x1234567890abcdef1234567890abcdef12345678901234567890abcdef123456",
            },
        ],
    },
]));
const mockOnrampCompletedStatuses = JSON.parse(stringify([
    {
        type: "onramp",
        status: "COMPLETED",
        transactions: [
            {
                chainId: 137,
                transactionHash: "0xabcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890",
            },
        ],
        purchaseData: {
            orderId: "stripe-order-abc123",
        },
    },
]));
// Wrapper component to provide theme context
const SuccessScreenWithTheme = (props) => {
    const { theme, ...componentProps } = props;
    return (_jsx(ModalThemeWrapper, { theme: theme, children: _jsx(SuccessScreen, { ...componentProps }) }));
};
const meta = {
    title: "Bridge/SuccessScreen",
    component: SuccessScreenWithTheme,
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "Success screen that displays completion confirmation with transaction summary, payment details, and action buttons for next steps. Includes animated success icon and detailed transaction view.",
            },
        },
    },
    tags: ["autodocs"],
    args: {
        preparedQuote: simpleBuyQuote,
        completedStatuses: mockBuyCompletedStatuses,
        onDone: () => { },
        theme: "dark",
        windowAdapter: webWindowAdapter,
        uiOptions: FUND_WALLET_UI_OPTIONS.ethDefault,
    },
    argTypes: {
        theme: {
            control: "select",
            options: ["light", "dark"],
            description: "Theme for the component",
        },
        onDone: { action: "success screen closed" },
    },
};
export default meta;
export const Default = {
    args: {
        theme: "dark",
    },
    parameters: {
        backgrounds: { default: "dark" },
    },
};
export const DefaultLight = {
    args: {
        theme: "light",
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
export const OnrampPayment = {
    args: {
        theme: "dark",
        preparedQuote: simpleOnrampQuote,
        completedStatuses: mockOnrampCompletedStatuses,
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Success screen for onramp payments showing payment ID that can be copied to clipboard.",
            },
        },
    },
};
export const OnrampPaymentLight = {
    args: {
        theme: "light",
        preparedQuote: simpleOnrampQuote,
        completedStatuses: mockOnrampCompletedStatuses,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
export const ComplexPayment = {
    args: {
        theme: "dark",
        preparedQuote: simpleOnrampQuote,
        completedStatuses: [
            ...mockOnrampCompletedStatuses,
            ...mockBuyCompletedStatuses,
        ],
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Success screen for onramp payments showing payment ID that can be copied to clipboard.",
            },
        },
    },
};
export const ComplexPaymentLight = {
    args: {
        theme: "light",
        preparedQuote: simpleOnrampQuote,
        completedStatuses: [
            ...mockOnrampCompletedStatuses,
            ...mockBuyCompletedStatuses,
        ],
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
export const TransactionPayment = {
    args: {
        theme: "light",
        preparedQuote: simpleBuyQuote,
        completedStatuses: mockBuyCompletedStatuses,
        uiOptions: TRANSACTION_UI_OPTIONS.contractInteraction,
    },
    parameters: {
        backgrounds: { default: "light" },
    },
};
