// thirdweb/dist/esm/stories/Bridge/UnsupportedTokenScreen.stories.js
import { jsx as _jsx } from "react/jsx-runtime";
import { defineChain } from "../../chains/utils.js";
import { UnsupportedTokenScreen, } from "../../react/web/ui/Bridge/UnsupportedTokenScreen.js";
import { ModalThemeWrapper } from "../utils.js";
// Wrapper component to provide theme context
const UnsupportedTokenScreenWithTheme = (props) => {
    const { theme, ...componentProps } = props;
    return (_jsx(ModalThemeWrapper, { theme: theme, children: _jsx(UnsupportedTokenScreen, { ...componentProps }) }));
};
const meta = {
    title: "Bridge/UnsupportedTokenScreen",
    component: UnsupportedTokenScreenWithTheme,
    parameters: {
        layout: "centered",
        docs: {
            description: {
                component: "Screen displayed when a token is being indexed or when using an unsupported testnet. Shows loading state for indexing tokens or error state for testnets.",
            },
        },
    },
    tags: ["autodocs"],
    args: {
        chain: defineChain(1), // Ethereum mainnet
        theme: "dark",
    },
    argTypes: {
        theme: {
            control: "select",
            options: ["light", "dark"],
            description: "Theme for the component",
        },
    },
};
export default meta;
export const TokenNotSupported = {
    args: {
        theme: "dark",
        chain: defineChain(1), // Ethereum mainnet - will show indexing spinner
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Shows the loading state when a token is being indexed by the Universal Bridge on a mainnet chain.",
            },
        },
    },
};
export const TokenNotSupportedLight = {
    args: {
        theme: "light",
        chain: defineChain(1), // Ethereum mainnet - will show indexing spinner
    },
    parameters: {
        backgrounds: { default: "light" },
        docs: {
            description: {
                story: "Shows the loading state when a token is being indexed by the Universal Bridge on a mainnet chain (light theme).",
            },
        },
    },
};
export const TestnetNotSupported = {
    args: {
        theme: "dark",
        chain: defineChain(11155111), // Sepolia testnet - will show error state
    },
    parameters: {
        backgrounds: { default: "dark" },
        docs: {
            description: {
                story: "Shows the error state when trying to use the Universal Bridge on a testnet chain (Sepolia in this example).",
            },
        },
    },
};
export const TestnetNotSupportedLight = {
    args: {
        theme: "light",
        chain: defineChain(11155111), // Sepolia testnet - will show error state
    },
    parameters: {
        backgrounds: { default: "light" },
        docs: {
            description: {
                story: "Shows the error state when trying to use the Universal Bridge on a testnet chain (Sepolia in this example, light theme).",
            },
        },
    },
};
