// thirdweb/dist/esm/react/web/ui/TransactionButton/TransactionModal.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useQuery } from "@tanstack/react-query";
import { useState } from "react";
import { trackPayEvent } from "../../../../analytics/track/pay.js";
import { resolvePromisedValue } from "../../../../utils/promise/resolve-promised-value.js";
import { CustomThemeProvider } from "../../../core/design-system/CustomThemeProvider.js";
import { useActiveAccount } from "../../../core/hooks/wallets/useActiveAccount.js";
import { useActiveWallet } from "../../../core/hooks/wallets/useActiveWallet.js";
import { webWindowAdapter } from "../../adapters/WindowAdapter.js";
import { LoadingScreen } from "../../wallets/shared/LoadingScreen.js";
import { BridgeOrchestrator } from "../Bridge/BridgeOrchestrator.js";
import { useConnectLocale } from "../ConnectWallet/locale/getConnectLocale.js";
import { Modal } from "../components/Modal.js";
import { DepositScreen } from "./DepositScreen.js";
import { ExecutingTxScreen } from "./ExecutingScreen.js";
export function TransactionModal(props) {
    const account = useActiveAccount();
    const wallet = useActiveWallet();
    useQuery({
        queryKey: ["transaction-modal-event", props.txId],
        queryFn: async () => {
            if (!account || !wallet) {
                throw new Error(); // never happens, because enabled is false
            }
            trackPayEvent({
                client: props.client,
                walletAddress: account.address,
                walletType: wallet.id,
                toChainId: props.tx.chain.id,
                toToken: props.tx.erc20Value
                    ? (await resolvePromisedValue(props.tx.erc20Value))?.tokenAddress
                    : undefined,
                event: props.modalMode === "buy"
                    ? "open_pay_transaction_modal"
                    : "open_pay_deposit_modal",
            });
            return null;
        },
        enabled: !!wallet && !!account,
    });
    return (_jsx(CustomThemeProvider, { theme: props.theme, children: _jsx(Modal, { open: true, size: "compact", setOpen: (_open) => {
                if (!_open) {
                    props.onClose();
                }
            }, children: _jsx(TransactionModalContent, { ...props }) }) }));
}
function TransactionModalContent(props) {
    const localeQuery = useConnectLocale(props.localeId);
    const [screen, setScreen] = useState("buy");
    if (!localeQuery.data) {
        return _jsx(LoadingScreen, {});
    }
    if (screen === "execute-tx") {
        return (_jsx(ExecutingTxScreen, { tx: props.tx, closeModal: props.onClose, onTxSent: props.onTxSent, windowAdapter: webWindowAdapter }));
    }
    if (props.modalMode === "deposit") {
        return (_jsx(DepositScreen, { client: props.client, onBack: props.onBack, tx: props.tx, connectLocale: localeQuery.data, onDone: () => {
                setScreen("execute-tx");
            } }));
    }
    return (_jsx(BridgeOrchestrator, { client: props.client, uiOptions: {
            mode: "transaction",
            transaction: props.tx,
        }, connectLocale: localeQuery.data, onComplete: () => {
            setScreen("execute-tx");
        }, onError: (_error) => { }, onCancel: props.onClose, connectOptions: undefined, receiverAddress: undefined, purchaseData: undefined, paymentLinkId: undefined, presetOptions: undefined }));
}
